### PR TITLE
Add user auth local cookie

### PR DESCRIPTION
**Pull request by cviecco**

### CODE REVIEW

File: broker/httpd/api.go [hunk 1]
```diff
@@ -42,10 +45,23 @@ func StartServer(staticConfig *staticconfiguration.StaticConfiguration, brokers
 		logger:       logger,
 		staticConfig: staticConfig,
 	}
+	// load templates
+	server.htmlTemplate = template.New("main")
+	/// Load the oter built in templates
```
Comment (rgooch) [inline broker/httpd/api.go]: s/oter/other/
Or perhaps s/oter/otter/ ?

File: broker/httpd/api.go [hunk 2]
```diff
@@ -42,10 +45,23 @@ func StartServer(staticConfig *staticconfiguration.StaticConfiguration, brokers
...
+			return nil, err
+		}
+	}
+
 	http.HandleFunc("/", server.rootHandler)
 	http.HandleFunc("/status", server.statusHandler)
 	serviceMux := http.NewServeMux()
-	serviceMux.HandleFunc("/", server.rootHandler)
+	serviceMux.HandleFunc("/", server.consoleAccessHandler)
+	serviceMux.HandleFunc("/static/", staticHandler)
```
Comment (rgooch) [inline broker/httpd/api.go]: Why the trailing slash?

File: broker/httpd/api.go [hunk 3]
```diff
@@ -42,10 +45,23 @@ func StartServer(staticConfig *staticconfiguration.StaticConfiguration, brokers
...
+	/// Load the oter built in templates
+	extraTemplates := []string{footerTemplateText, consoleAccessTemplateText, headerTemplateText}
+	for _, templateString := range extraTemplates {
+		_, err = server.htmlTemplate.Parse(templateString)
+		if err != nil {
+			return nil, err
+		}
+	}
+
 	http.HandleFunc("/", server.rootHandler)
```
Comment (rgooch) [inline broker/httpd/api.go]: s/rootHandler/dashboardRootHandler/ ?

File: broker/httpd/serviceEndpoints.go [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+	"fmt"
+	"net/http"
+)
+
+func staticHandler(w http.ResponseWriter, r *http.Request) {
+	//log.Printf("Top of Static Handler")
+	switch r.URL.Path {
+	case "/static/common.css":
+		w.Header().Set("Content-Type", "text/css")
+		fmt.Fprintf(w, "%s", commonCSS)
```
Comment (rgooch) [inline broker/httpd/serviceEndpoints.go]: Please use `io.WriteString` instead. Same for below.

File: broker/httpd/serviceEndpoints.go [hunk 5]
```diff
@@ -0,0 +1,38 @@
+package httpd
+
+import (
+	"fmt"
+	"net/http"
+)
+
+func staticHandler(w http.ResponseWriter, r *http.Request) {
+	//log.Printf("Top of Static Handler")
```
Comment (rgooch) [inline broker/httpd/serviceEndpoints.go]: Please remove comment.

File: broker/httpd/serviceEndpoints.go [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+	case "/static/customization.css":
+		w.Header().Set("Content-Type", "text/css")
+		fmt.Fprintf(w, "%s", customizationCSS)
+		return
+	}
+	http.Error(w, "error not found", http.StatusNotFound)
+}
+
+func (s *Server) consoleAccessHandler(w http.ResponseWriter, r *http.Request) {
+
```
Comment (rgooch) [inline broker/httpd/serviceEndpoints.go]: Please remove blank line.

File: broker/httpd/serviceEndpoints.go [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+	}
+	err := s.htmlTemplate.ExecuteTemplate(w, "consoleAccessPage", displayData)
+	if err != nil {
+		s.logger.Printf("Failed to execute %v", err)
+		http.Error(w, "error", http.StatusInternalServerError)
+		return
+	}
+	return
+
+	//consoleAccessTemplateText
```
Comment (rgooch) [inline broker/httpd/serviceEndpoints.go]: Please remove blank line and comment.

**Review by rgooch — CHANGES_REQUESTED** [review]

File: broker/httpd/api.go (hunk 1 above)
Comment (zoresvit) [reply broker/httpd/api.go]: otter

File: broker/httpd/api.go (hunk 2 above)
Comment (cviecco) [reply broker/httpd/api.go]: I like terminating slashes for clarity.

File: broker/httpd/api.go [hunk 8]
```diff
@@ -47,7 +47,7 @@ func StartServer(staticConfig *staticconfiguration.StaticConfiguration, brokers
 	}
 	// load templates
 	server.htmlTemplate = template.New("main")
-	/// Load the oter built in templates
+	/// Load the other built in templates
```
Comment (rgooch) [inline broker/httpd/api.go]: Two slashes, please, not three.

File: broker/httpd/api.go (hunk 2 above)
Comment (rgooch) [reply broker/httpd/api.go]: Hm. OK for HTTP URLs, I guess. Can you please make this consistent, then?

**Review by rgooch — CHANGES_REQUESTED** [review]

**Pull request merged**